connector/github/github.go

```diff
@@ -30,17 +35,46 @@ type Config struct {
 	ClientSecret string `json:"clientSecret"`
 	RedirectURI  string `json:"redirectURI"`
 	Org          string `json:"org"`
+	HostName     string `json:"hostName"`
+	RootCA       string `json:"rootCA"`
 }
 
 // Open returns a strategy for logging in through GitHub.
 func (c *Config) Open(logger logrus.FieldLogger) (connector.Connector, error) {
-	return &githubConnector{
+	g := githubConnector{
 		redirectURI:  c.RedirectURI,
 		org:          c.Org,
 		clientID:     c.ClientID,
 		clientSecret: c.ClientSecret,
+		apiURL:       apiURL,
 		logger:       logger,
-	}, nil
+	}
+
+	if c.HostName != "" {
+		// make sure this is a hostname and not a URL.
+		// Ensure it doesn't have a schema or a path.
+		if _, err := url.ParseRequestURI(c.HostName); err == nil {
```

Comment:
I think `url.Parse` and `url.ParseRequestURI` are the wrong tools here since they can both produce false positives (e.g. say something is a URL when it isnt').

Ideally we'd have a `IsHostname` function. But lacking that, we should be conservative about the validation.